services/src/main/java/org/fao/geonet/api/ApiUtils.java

```diff
@@ -470,13 +474,35 @@ static public AbstractMetadata canChangeStatusRecord(String metadataUuid, Servic
      * @throws SecurityException if user is not allowed to view
      */
     public static AbstractMetadata canViewRecord(String metadataUuid, HttpServletRequest request) throws Exception {
+        return canViewRecord(metadataUuid, true, request);
+    }
+    /**
+     * Check if the current user can view this record.
+     *
+     * This method creates a temporary service context using the provided request to check record access,
+     * if you have a service context already please use {@link #canViewRecord(String, ServiceContext)}.
+     *
+     * @param metadataUuid Look up metadata record
+     * @param request Request to identify current user
+     * @return metadata record
+     * @throws SecurityException if user is not allowed to view
+     */
+    public static AbstractMetadata canViewRecord(String metadataUuid, boolean approved, HttpServletRequest request) throws Exception {
+        String metadataId;
+        if (!approved) {
+            // If the record is not approved then we need to get the id of the record.
+            metadataId = getInternalId(metadataUuid, approved);
+        } else {
+            // Otherwise use the uuid or id that was supplied.
+            metadataId = metadataUuid;
+        }
```

Comment:
```suggestion
        metadataId = getInternalId(metadataUuid, approved);
```

Reply:
Note that the reason I did not do this in one statement was due to performance.
If it is an approved record, it would only call 
`canViewRecord(metadataId, context);`
But by merging your changes, it will always call `getInternalId` and then `canViewRecord(metadataId, context);`

I don't expect `getInternalId` to cause much performance issues so I merged your changes